### PR TITLE
feat: add Os and Arch template options to the Binary name field

### PR DESCRIPTION
**Pull request by sylviamoss**
If applied, this PR will add the capability to use `{{ .Os }}` and `{{.Arch}}` in the binary name. 

I'm implementing this change because I was expecting these to work at this point of the build process. I'm using this specific configuration for the binary name, to make it easier for [Packer](https://github.com/hashicorp/packer) when managing plugin's binaries for different os/arch without needing to add the information to the binary name itself afterward.

Here a failing release example for a repository I'm using for testing:
https://github.com/sylviamoss/packer-provisioner-comment/runs/1511229354?check_suite_focus=true#step:6:58

I tested the changes locally and here's the output line with the binary name: 
```
   • building binaries
      • building                  binary=/Users/sylviamoss/Packer/plugins-example/packer-provisioner-comment/dist/packer-provisioner-comment_freebsd_amd64/packer-provisioner-comment_v0.2.2-SNAPSHOT-a1d03f2_freebsd_amd64
```

### CODE REVIEW

**Comment by caarlos0** [comment]
I think you would want to do that by using `archive.format: binary` and setting the templates there... does that make sense?

**Comment by sylviamoss** [comment]
I'm creating zip archives. The way our tool works is by keeping the binaries in a folder to use them at the runtime. One can have binaries for different os/arch for the same plugin and that's why we differentiate them by their names. We are parsing the binary name and adding these information to it, it would be nice if they were created already with them.

**Review by caarlos0 — COMMENTED** [review]
LGTM overall, can you update the docs too? (`www/docs/customization/build.md`)

**Comment by sylviamoss** [comment]
@caarlos0 I updated the documentation but rebased with master and messed everything up. 🤦🏼

**Review by radeksimko — APPROVED** [review]
LGTM.

I don't have access to Vercel to verify how the markdown is rendered, but it looks good.

File: www/docs/customization/build.md [hunk 1]
```diff
@@ -159,6 +159,17 @@ builds:
       - windows
 ```
 
+The binary name field supports [templating](/customization/templates/). The following build details are exposed:
+
+| Key     | Description                      |
+|---------|----------------------------------|
+| .Os     | `GOOS`                           |
+| .Arch   | `GOARCH`                         |
+| .Target | `GOARM`                          |
```
Comment (caarlos0) [inline www/docs/customization/build.md:168]: this doesn't seem correct...

Comment (sylviamoss) [reply www/docs/customization/build.md:168]: that's true, I did some confusion here 🤦🏼 
fixing it now

**Review by caarlos0 — APPROVED** [review]

**Comment by caarlos0** [comment]
LGTM, thanks!

**Pull request merged**